module/zfs/zcp_get.c

```diff
@@ -411,6 +411,11 @@ get_special_prop(lua_State *state, dsl_dataset_t *ds, const char *dsname,
 	case ZFS_PROP_INCONSISTENT:
 		numval = dsl_get_inconsistent(ds);
 		break;
+	case ZFS_PROP_IVSET_GUID:
+		error = zap_lookup(ds->ds_dir->dd_pool->dp_meta_objset,
+		    ds->ds_object, DS_FIELD_IVSET_GUID, sizeof (numval),
+		    1, &numval);
```

Comment:
Doing this without checking if the dataset is zapified won't work right, for a few reasons:
1. on debug bits, this will panic (see zap_lockdir())
2. on nondebug bits on other platforms (e.g. illumos), this will probably panic, since the check in zap_lockdir_impl() is unique to ZoL.
3. on nondebug bits on ZoL, this will return EINVAL (incorrect channel program) when we want ENOENT (property not present).  See zcp_handle_error().